### PR TITLE
Add metadata_props to ORT model

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: Add metadata_props to ORT model

**Motivation and Context**
- Request from 1p user for metadata_props in ort model
- Fixed a small bug in serializing model_version
- Added UT test case
- Since this change is backward compatible, (old version of ORT can read the .ort file generated in this change without issue, only the metadata_props will be empty, and this field is not critical), we will not bump the ORT file version in this case.

### CODE REVIEW

File: onnxruntime/test/framework/ort_model_only_test.cc [hunk 1]
```diff
@@ -193,6 +193,26 @@ static void CompareGraphAndSessionState(const InferenceSessionWrapper& session_o
...
+  ASSERT_STATUS_OK(pair_1.first);
+  const auto& metadata_1 = *pair_1.second;
+
+  const auto pair_2 = session_object_2.GetModelMetadata();
+  ASSERT_STATUS_OK(pair_2.first);
+  const auto& metadata_2 = *pair_2.second;
+
+  ASSERT_EQ(metadata_1.producer_name, metadata_2.producer_name);
+  // ORT format does not have graph name
+  // ASSERT_EQ(metadata_1.graph_name, metadata_2.graph_name);
```
Comment (gwang-msft) [inline onnxruntime/test/framework/ort_model_only_test.cc:208]: Do we want to add this also?

Comment (skottmckay) [reply onnxruntime/test/framework/ort_model_only_test.cc:208]: Given we can add it if needed in the future without breaking backwards compat, I'd hold off on adding it. 

If someone is desperate they could use the graph description as a fallback.

File: onnxruntime/core/flatbuffers/schema/ort.fbs [hunk 2]
```diff
@@ -202,10 +207,11 @@ table Model {
...
   model_version:int64;
   doc_string:string;
-  
+
   graph:Graph;
-  
+
   graph_doc_string:string;
+  metadata_props:[StringStringEntry];
 }
```
Comment (skottmckay) [inline onnxruntime/core/flatbuffers/schema/ort.fbs:215]: Can you regenerate the python bindings as well? 

In theory, you can just run \onnxruntime\core\flatbuffers\create_python_bindings.py to handle that.

File: onnxruntime/core/graph/model.cc [hunk 3]
```diff
@@ -703,6 +731,13 @@ common::Status Model::LoadFromOrtFormat(const fbs::Model& fbs_model,
       schema_registry->RegisterRegistry(schema_collection);
     }
   }
+
+  // Populate the metadata to model_proto
```
Comment (gwang-msft) [inline onnxruntime/core/graph/model.cc:735]: Not sure of this part is necessary, only thing I can think of is to convert an ort model back to onnx format, maybe we can remove this to save a bit binary size.

File: onnxruntime/core/flatbuffers/schema/ort.py.fbs [hunk 4]
```diff
@@ -0,0 +1,245 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace ort_flatbuffers_py.experimental.fbs;
+
```
Comment (skottmckay) [inline onnxruntime/core/flatbuffers/schema/ort.py.fbs:5]: I didn't check this in previously as I considered it to be a temporary hack to generated the python bindings with a usable namespace. Checking it in might also be confusing - e.g. should you edit this and the ort.fbs file, and how are the namespaces for each handled.

Is there a need or reason to check it in now?

Comment (gwang-msft) [reply onnxruntime/core/flatbuffers/schema/ort.py.fbs]: Added a .gitignore here to prevent this to happen in the future

**Review by skottmckay — APPROVED** [review]
:shipit:

File: onnxruntime/core/graph/model.cc [hunk 5]
```diff
@@ -686,6 +702,18 @@ common::Status Model::LoadFromOrtFormat(const fbs::Model& fbs_model,
                                         std::unique_ptr<Model>& model) {
   model.reset(new Model());
 
+  // Load the model metadata
+  if (const auto* fbs_metadata_props = fbs_model.metadata_props()) {
+    model->model_metadata_.reserve(fbs_metadata_props->size());
+    for (const auto* prop : *fbs_metadata_props) {
+      ORT_RETURN_IF(nullptr == prop, "Null entry in meta_props. Invalid ORT format model.");
```
Comment (edgchen1) [inline onnxruntime/core/graph/model.cc:709]:
```suggestion
      ORT_RETURN_IF(nullptr == prop, "Null entry in metadata_props. Invalid ORT format model.");
```

File: tools/ci_build/github/linux/ort_minimal/build_minimal_ort_android_baseline_and_report_bin_size.sh [hunk 6]
```diff
@@ -30,7 +30,7 @@ python3 /onnxruntime_src/tools/ci_build/build.py \
 
 # set current size limit to 1165KB.
```
Comment (edgchen1) [inline tools/ci_build/github/linux/ort_minimal/build_minimal_ort_android_baseline_and_report_bin_size.sh:31]:
```suggestion
# set current size limit to 1175KB.
```
or replace this comment with something like `BINARY_SIZE_LIMIT_IN_BYTES=1175000` and use that below

**Review by edgchen1 — APPROVED** [review]

File: onnxruntime/core/graph/model.cc (hunk 5 above)
Comment (gwang-msft) [reply onnxruntime/core/graph/model.cc:709]: Will fix these comments in my next PR

File: tools/ci_build/github/linux/ort_minimal/build_minimal_ort_android_baseline_and_report_bin_size.sh (hunk 6 above)
Comment (gwang-msft) [reply tools/ci_build/github/linux/ort_minimal/build_minimal_ort_android_baseline_and_report_bin_size.sh:31]: Will fix these comments in my next PR

**Pull request merged**